src/wp-includes/blocks.php

```diff
@@ -760,6 +760,108 @@ function get_hooked_blocks( $name ) {
 	return $hooked_blocks;
 }
 
+function insert_hooked_blocks( $block ) {
+	$hooked_blocks = get_hooked_blocks( $block['blockName'] );
+	foreach ( $hooked_blocks as $hooked_block_type => $relative_position ) {
+		$hooked_block = array(
+			'blockName'    => $hooked_block_type,
+			'attrs'        => array(),
+			'innerHTML'    => '',
+			'innerContent' => array(),
+			'innerBlocks'  => array(),
+		);
+		// Need to pass full current block, (possibly its parent), relative position, and hooked_block.
+		$block = insert_hooked_block( $hooked_block, $relative_position, $block );
+	}
+	return $block;
+}
+
+/**
+ * Auto-inserts a block next to a given "anchor" block.
+ *
+ * This is a helper function used in the implementation of block hooks.
+ * It is not meant for public use.
+ *
+ * The auto-inserted block can be inserted before or after the anchor block,
+ * or as the first or last child of the anchor block.
+ *
+ * Note that the this mutates the automatically inserted block's
+ * designated parent block by inserting into the parent's `innerBlocks` array,
+ * and by updating the parent's `innerContent` array accordingly.
+ *
+ * @since 6.4.0
+ *
+ * @param array  $parent_block      The parent block of the automatically inserted block.
+ * @param array  $inserted_block    The block to insert.
+ * @param string $relative_position The position relative to the given anchor block.
+ *                                  Can be 'before', 'after', 'first_child', or 'last_child'.
+ * @param array $anchor_block       The automatically inserted block will be inserted next to instances of this block type.
+ *                                  Only needed when inserting before or after the anchor block.
+ * @return array The modified anchor block.
+ */
+function insert_hooked_block( $parent_block, $inserted_block, $relative_position, $anchor_block = null ) {
+	if ( 'first_child' === $relative_position ) {
+		array_unshift( $parent_block['innerBlocks'], $inserted_block );
+		/*
+		 * Since WP_Block::render() iterates over `inner_content` (rather than `inner_blocks`)
+		 * when rendering blocks, we also need to prepend a value (`null`, to mark a block
+		 * location) to that array (but after the HTML content for the inner blocks wrapper).
+		 */
+		$chunk_index = 0;
+		for ( $index = $chunk_index; $index < count( $parent_block['innerContent'] ); $index++ ) {
+			if ( is_null( $parent_block['innerContent'][ $index ] ) ) {
+				$chunk_index = $index;
+				break;
+			}
+		}
+		array_splice( $parent_block['innerContent'], $chunk_index, 0, array( null ) );
+		return $parent_block;
+	} elseif ( 'last_child' === $relative_position ) {
```

Comment:
No need to use else anymore because there is the `return` statement:
```suggestion
	}
	if ( 'last_child' === $relative_position ) {
```